snapcraft/plugins/nodejs.py

```diff
@@ -62,18 +64,23 @@ def schema(cls):
             'items': {
                 'type': 'string'
             },
-            'default': [],
+            'default': []
         }
-
-        if 'required' in schema:
-            del schema['required']
+        schema['properties']['node-engine'] = {
+            'type': 'string'
+        }
+        # `node-engine` keyword is required, but `node-packages` is not.
+        schema['required'].append('node-engine')
+        schema['required'].remove('source')
+        schema['properties']['source']['default'] = '.'
```

Comment:
I doubt you want this line here (defaulting the source to '.'). Also, the comment doesn't seem quite right. Perhaps:

```python
# `node-engine` keyword is required, but `source` is not.
schema['required'].append('node-engine')
schema['required'].remove('source')
```